src/app/archive/lib/processor.ml

```diff
@@ -447,6 +600,30 @@ module Zkapp_states = struct
         ; element5
         ; element6
         ; element7
+        ; element8
+        ; element9
+        ; element10
+        ; element11
+        ; element12
+        ; element13
```

Comment:
Why is it not a check on length of the list? May I ask?

Reply:
I think because you need to bring `element0..element31` into scope anyway you might as well match like this.
AFAICT the root cause of this ugly code is that we use structs for fixed width arrays in SQL.

https://github.com/MinaProtocol/mina/blob/4cad6b75b611ee520645b74d0fad8506b662ec00/src/app/archive/create_schema.sql#L138-L153

Maybe we should just use arrays and only enforce the length on the ocaml side?

Reply:
Actually I think I'm wrong about that. I'll look into this more if I have time.